src/chessboard/components/Piece.tsx

```diff
@@ -90,7 +90,10 @@ export function Piece({
     if (premoves.find((p) => p.sourceSq === square && p.piece === piece))
       hidePiece = true;
 
-    // TODO: If a premoved piece returns to a premoved square, it will hide (e1, e2, e1)
+    // The last premoved piece should always be visible
+    if (isPremovedPiece && premoves.at(-1)?.targetSq === square) {
+      hidePiece = false;
+    }
```

Comment:
So if someone premoves a piece, then premoves it back to its original position then this will work

However, what if someone then premoves a different piece? It will hide again I think right?

Reply:
hmmmmm 🤔, you are right

Reply:
fix looks not as easy as I imagined, lets say we do many premoves with both of our knights and they sometimes move to the exact square. In that case,  figuring out which piece where have been moved is too hard, so I suggest inventing some inner unique ID for each piece on the chessboard. It can also be useful in the future for other scenarios I guess.